src/ynEIGEN/ynEigenViewer.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: BSD 3-Clause License
+pragma solidity ^0.8.24;
+
+import {IERC20, IERC20Metadata} from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+
+import {IynEigen} from "../interfaces/IynEigen.sol";
+import {IRateProvider} from "../interfaces/IRateProvider.sol";
+import {ITokenStakingNodesManager,ITokenStakingNode} from "../interfaces/ITokenStakingNodesManager.sol";
+
+import {AssetRegistry} from "./AssetRegistry.sol";
+
+contract ynEigenViewer {
+    
+    struct AssetInfo {
+        address asset;
+        string name;
+        string symbol;
+        uint256 ratioOfTotalAssets;
+        uint256 totalBalance;
+    }
+    
+    AssetRegistry public immutable assetRegistry;
+    IynEigen public immutable ynEIGEN;
+    ITokenStakingNodesManager public immutable tokenStakingNodesManager;
+    IRateProvider public immutable rateProvider;
+
+    uint256 public constant DECIMALS = 1_000;
```

Comment:
lets' make this 10_000;

Reply:
https://github.com/yieldnest/yieldnest-protocol/pull/145/commits/58a1e20c98aab3d15ce52a0faa07d403ccdb2a49